oauthproxy.go

```diff
@@ -351,7 +366,7 @@ func buildSessionChain(opts *options.Options, sessionStore sessionsapi.SessionSt
 
 	if opts.SkipJwtBearerTokens {
 		sessionLoaders := []middlewareapi.TokenToSessionFunc{
-			opts.GetProvider().CreateSessionFromToken,
+			opts.GetProvider(0).CreateSessionFromToken,
```

Comment:
currently to preserve functionality the s2s flow will use provider [0] - but as i understand it oidc tokens don't actually care about the provider and will just check the custom jwt issuer URLs 

@NickMeves do you have any suggestions on how we can integrate this a little bit better?